R/bettr-main.R

```diff
@@ -31,63 +55,95 @@
 #'   actionButton br tabPanel titlePanel navbarPage
 #' @importFrom sortable rank_list
-#' @importFrom ggplot2 theme_minimal geom_bar aes ggplot geom_boxplot 
-#'   coord_flip geom_jitter 
+#' @importFrom shinyjqui jqui_resizable
 #' @importFrom dplyr select %>% contains
-#' @importFrom cowplot plot_grid
 #' @importFrom tidyr gather
 #' @importFrom bslib bs_theme
 #' @importFrom rlang .data
 #' 
+#' @examples 
+#' df <- data.frame(Method = c("M1", "M2", "M3"), metric1 = c(1, 2, 3),
+#'                  metric2 = c(3, 1, 2), metric3 = factor(c("a", "a", "b")))
+#' initialTransforms <- list(metric1 = list(flip = TRUE, offset = 4))
+#' metricInfo <- data.frame(Metric = c("metric1", "metric2", "metric3"),
+#'                          Group = c("G1", "G2", "G2"))
+#' idInfo <- data.frame(Method = c("M1", "M2", "M3"), 
+#'                      Type = c("T1", "T1", "T2"))
+#' metricColors = list(Group = c(G1 = "red", G2 = "blue"))
+#' if (interactive()) {
+#'     bettr(df = df, idCol = "Method", 
+#'     metrics = c("metric1", "metric2", "metric3"),
+#'     initialTransforms = initialTransforms,
+#'     metricInfo = metricInfo, metricColors = metricColors,
+#'     idInfo = idInfo)
+#' }
+#' 
 bettr <- function(df, idCol = "Method", 
-                  metrics_num = setdiff(colnames(df), idCol),
-                  metrics_cat = c(), initialWeights = NULL,
-                  initialFlips = NULL, initialOffsets = NULL,
-                  initialTransforms = NULL, initialCuts = NULL,
-                  metricGroups = list(), bstheme = "darkly") {
-    
-    ## All metrics (numeric and categorical) ----------------------------------
-    metrics <- c(metrics_num, metrics_cat)
-    
-    ## Check validity of input arguments --------------------------------------
-    .checkInputArguments(df = df, idCol = idCol, metrics_num = metrics_num,
-                         metrics_cat = metrics_cat, 
-                         initialWeights = initialWeights,
-                         initialFlips = initialFlips,
-                         initialOffsets = initialOffsets,
-                         initialTransforms = initialTransforms,
-                         initialCuts = initialCuts,
-                         metricGroups = metricGroups, bstheme = bstheme)
+                  metrics = setdiff(colnames(df), idCol),
+                  initialWeights = NULL,
+                  initialTransforms = list(),
+                  metricInfo = NULL, metricColors = NULL,
+                  idInfo = NULL, idColors = NULL,
+                  bstheme = "darkly") {
     
-    ## Define column names assigned internally ---- ---------------------------
+    ## Define column names assigned internally --------------------------------
     scoreCol <- "Score"
     weightCol <- "Weight"
     metricCol <- "Metric"
     valueCol <- "ScaledValue"
     groupCol <- "Group"
+    initialWeightValue <- 0.2
+    weightResolution <- 0.05
```

Comment:
Maybe overkill, but - shall we promote these as well to "main" arguments of the function? In case one wants to tweak these, and we keep these sensible defaults

Reply:
For the resolution - sure, can add as an argument. For the `initialWeightValue`, I think it would be a bit confusing. You can already specify the initial weights, and this is the value that will be used if you don't say anything, or when you want to 'reset to uniform weights'. At that point, it doesn't really matter what the value is, in my opinion.

Reply:
I thought of it as a mix of "time saver" feature, and on top of it, I saw a hard variable assignment and thought it should not belong there :D